/**
 * The MIT License (MIT)
 * Copyright (c) 2016-2017 Intel Corporation
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy of 
 * this software and associated documentation files (the "Software"), to deal in 
 * the Software without restriction, including without limitation the rights to 
 * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of 
 * the Software, and to permit persons to whom the Software is furnished to do so, 
 * subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all 
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS 
 * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR 
 * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER 
 * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
*/

/**
* Note: Variable names are kept aligned with
* GenomicsDB JSON configuration.
*
* To build, use protocol buffer version >3.2.0 and the following command from base directory:
* $ protoc -Isrc/resources/ --java_out=src/main/java/ \
*     src/resources/genomicsdb_callsets_mapping.proto
*/

syntax = "proto2";

option java_package = "com.intel.genomicsdb.model";
option java_outer_classname = "GenomicsDBCallsetsMapProto";

message SampleIDToTileDBIDMap {
  required string sample_name = 1;
  required int64 row_idx = 2;
  required int64 idx_in_file = 3;
  optional string stream_name = 4;
}

message CallsetMappingPB {
  repeated SampleIDToTileDBIDMap callsets = 1;
}
